File: defanging-an-ip-address/DefangingAnIpAddress.cs
using System;

namespace LeetCode
{
    class DefangingAnIpAddress
    {
        public String DefangingAnIpAddressSolution(String address)
        {
            var result = "";
            foreach(char character in address)
            {
                if(character == '.')
                {
                    result += "[.]";
                }
                else
                {
                    result += character;
                }
            }

            return result;
        }
    }
}

File: defanging-an-ip-address/DefangingAnIpAddress.java

public class DefangingAnIpAddress {

	public String DefangingAnIpAddressSolution(String address) {
		String result = new String("");
		for (char c: address.toCharArray()) {
			if (c == '.')
			{
				result += "[.]";
			} else {
				result += c;
			}
		}
		return result;
	}
}
